### ISSUE TITLE
Coding guidelines

### ISSUE THREAD

**Issue by Christian24** [viadee/vPAV#72]
Right now we do not have coding guidelines. I created a first draft in our wiki here: https://github.com/viadee/vPAV/wiki/Coding-guidelines

It's a living document, so please join the discussion. I think we should also name a Code Formatter that every  contributor should use.

**Comment by fkoehne** [comment]
Definately a good idea. But I don't think we should discuss and document this locally. I would prefer to just provide a pointer to an existing set of guidelines such as these https://google.github.io/styleguide/javaguide.html.

**Comment by Christian24** [comment]
I agree. However, I am afraid we would put too much of a burden on contributing. If I want to contribute I will probably not read the entire style guide (if it is that long). 

Maybe we should (only) refer to the formatting plugins: https://github.com/google/google-java-format

**Comment by fkoehne** [comment]
Let's give it a try. :)

**Comment by Christian24** [comment]
I updated the coding guidelines to reflect the usage of the Google Java Style Guide. And I tried the IntelliJ plugin and am very pleased with it. 

We need to reformat the existing codebase though.

**Comment by fkoehne** [comment]
... with the maven plugin?

**Comment by Christian24** [comment]
Have you tried any of the three?

**Comment by fkoehne** [comment]
Not yet - sorry.

**Comment by chrlembeck** [comment]
If you want to customize your Java Style Guide, there is another maven plugin based on the eclipse code formatter: https://github.com/revelc/formatter-maven-plugin

**Comment by Christian24** [comment]
I have just pushed an updated contributing branch with updated contributing guidelines and Java code formatter applied to all source files. 

What do you think? To me it looks a bit like the indention is really small.

**Comment by sdibernardo** [comment]
I have to agree that I am not a fan of the indent being that small. Besides that, I have not found any other concerns

**Comment by Christian24** [comment]
Well, I had a second thought on this: If the indentation is this small, maybe developers no longer nest this deeply. :) Maybe we should just run with it for now, since so far the advantage of properly formatted code by far outweighs the inconvenience of not being able to nest so deeply so easily for me. If we come across a better formatter, we can still switch later on.

**Comment by fkoehne** [comment]
Sure - go for it. Please make sure that the setup is easily reproducable with both IDEs at hand and well documented :).

**Comment by Christian24** [comment]
I would argue that Google did that for us already: https://github.com/google/google-java-format

**Comment by fkoehne** [comment]
Well - yes. But that was not the point I wanted to make. Let's make noticing this and hopping on as a contributor as easy as we can.
